File: ZJH_Demo组件化的三种方式/ZJH_Demo/Controller/ViewController.m
//
//  ViewController.m
//  ZJH_Demo
//
//  Created by keloopA on 2024/2/20.
//

#import "ViewController.h"
#import "Meditor.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}
- (IBAction)jumpToDetail:(UIButton *)sender {
    
//    UIViewController* vc = [Meditor detailViewController];
//    [self.navigationController pushViewController:vc animated:YES];
    
//    [Meditor openDetailVCWithSchem:@"detail://"];
    
    Class cls = [Meditor getClassWithProtocol:@protocol(DetailViewControllerProtocol)];
    [cls showDetail];
    
}


@end

File: NSCoder/NSCoder/ViewController.swift
//
//  ViewController.swift
//  NSCoder
//
//  Created by keloopA on 2024/2/21.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        let p = Person()
       let data = try! NSKeyedArchiver.archivedData(withRootObject: p, requiringSecureCoding: true)
        
       let res = try! NSKeyedUnarchiver.unarchivedObject(ofClass: Person.self, from: data)
        print(res)
     
    }
}

